assets/js/googlesitekit/datastore/site/first-party-mode.js

```diff
@@ -0,0 +1,226 @@
+			useCache: false,
+		} ),
+	reducerCallback: settingsReducerCallback,
+} );
+
+const fetchSaveFirstPartyModeSettingsStore = createFetchStore( {
+	baseName: 'saveFirstPartyModeSettings',
+	controlCallback: ( { settings } ) => {
+		return API.set( 'core', 'site', 'fpm-settings', { settings } );
+	},
+	reducerCallback: settingsReducerCallback,
+	argsToParams: ( settings ) => {
+		const { isEnabled } = settings || {};
+		return { settings: { isEnabled } };
+	},
+	validateParams: ( { settings } ) => {
+		invariant(
+			isPlainObject( settings ),
+			'settings must be a plain object.'
+		);
+
+		invariant(
+			typeof settings.isEnabled === 'boolean',
+			'isEnabled must be a boolean.'
+		);
+
+		invariant(
+			Object.keys( settings ).length === 1,
+			'settings must have only the `isEnabled` property.'
+		);
+	},
+} );
+
+const baseInitialState = {
+	firstPartyModeSettings: undefined,
+};
+
+const baseActions = {
+	/**
+	 * Saves the first-party mode settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*saveFirstPartyModeSettings() {
+		const { select } = yield commonActions.getRegistry();
+		const settings = select( CORE_SITE ).getFirstPartyModeSettings();
+
+		return yield fetchSaveFirstPartyModeSettingsStore.actions.fetchSaveFirstPartyModeSettings(
+			settings
+		);
+	},
+
+	/**
+	 * Sets the first-party mode enabled status.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {boolean} isEnabled First-party mode enabled status.
+	 * @return {Object} Redux-style action.
+	 */
+	setFirstPartyModeEnabled( isEnabled ) {
+		return {
+			type: SET_FIRST_PARTY_MODE_ENABLED,
+			payload: { isEnabled },
+		};
+	},
+};
+
+const baseControls = {};
+
+const baseReducer = createReducer( ( state, { type, payload } ) => {
+	switch ( type ) {
+		case SET_FIRST_PARTY_MODE_ENABLED: {
+			state.firstPartyModeSettings = state.firstPartyModeSettings || {};
+			state.firstPartyModeSettings.isEnabled = !! payload.isEnabled;
+			break;
+		}
+
+		default:
+			break;
+	}
+} );
+
+const baseResolvers = {
+	*getFirstPartyModeSettings() {
+		const { select } = yield commonActions.getRegistry();
+
+		const settings = select( CORE_SITE ).getFirstPartyModeSettings();
+
+		if ( settings === undefined ) {
+			yield fetchGetFirstPartyModeSettingsStore.actions.fetchGetFirstPartyModeSettings();
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Gets the first-party mode settings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} First-party mode settings, or undefined if not loaded.
+	 */
+	getFirstPartyModeSettings: ( state ) => {
+		return state.firstPartyModeSettings;
+	},
+
+	/**
+	 * Checks if first-party mode is enabled.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {boolean|undefined} True if first-party mode is enabled, otherwise false. Returned undefined if the state is not loaded.
```

Comment:
```suggestion
	 * @return {boolean|undefined} True if first-party mode is enabled, otherwise false. Returns undefined if the state is not loaded.
```